core/src/main/scala/org/apache/spark/util/SignalUtils.scala

```diff
@@ -31,14 +31,51 @@ import org.apache.spark.internal.Logging
 /**
  * Contains utilities for working with posix signals.
  */
-private[spark] object Signaling extends Logging {
+private[spark] object SignalUtils extends Logging {
+
+  private var registered = false
+
+  /** Register a signal handler to log signals on UNIX-like systems. */
+  def registerLogger(log: Logger): Unit = synchronized {
```

Comment:
yea but it doesn't make sense to have that class there hanging by itself with a tiny bit of functionality